methods.rb

```diff
@@ -0,0 +1,116 @@
+module Enumerable
+  def my_each
+    i = 0
+    length.times do
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_map(proc = nil)
+    temp = []
+    i = 0
+    if proc.nil?
+      length.times do
+        temp.push(yield(self[i]))
+        i += 1
+      end
+    else
+      length.times do
+        temp.push(proc.call(self[i]))
+        i += 1
+      end
+    end
+    temp
+  end
```

Comment:
- [ ] `my_map` is not working as the original `map` method, also is not passing all test cases.

### `map` is an Enumerable that has these characteristics:
- Returns an Enumerator if no block is given.
```
[1, 2, 3].map
# #<Enumerator: [1, 2, 3]:map>
```
- Returns a **new array** with the results of running a block
```
[1, 2, 3].map { |num| num * 2 }
# [2, 4, 6]
```
- doesn't change/modify the original array/range/hash
- can accept a proc as an argument
```
['1', '2', '3'].map(&:to_i)
# [1, 2, 3]
```
[Enumerable#map](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-map)